libs/application-generic/src/services/support.service.ts

```diff
@@ -5,12 +5,11 @@ const LOG_CONTEXT = 'SupportService';
 
 export class SupportService {
   private plainClient: PlainClient;
-  constructor(private plainKey?: string | null) {
-    if (this.plainKey) {
-      this.plainClient = new PlainClient({
-        apiKey: this.plainKey,
-      });
-    }
+  private readonly plainKey: string;
+  constructor() {
+    this.plainKey = process.env.PLAIN_SUPPORT_KEY;
+    this.plainClient = new PlainClient({ apiKey: this.plainKey });
```

Comment:
How does the constructor behave when `PLAIN_SUPPORT_KEY` is _not_ present? Does this throw an error?

If it errors, the community edition will face issues. I would suggest only instantiating the client when `plainKey` is present in the environment. Then, conditionally access the upsert method like `this.plainClient?.upsertCustomer` in the upsert method